core/templates/components/state-editor/state-responses-editor/state-responses.component.ts

```diff
@@ -143,19 +142,23 @@ export class StateResponsesComponent implements OnInit, OnDestroy {
     this.stateSolicitAnswerDetailsService.saveDisplayedValue();
   }
 
+  getActiveStateName(): string | null {
+    return this.stateEditorService.getActiveStateName();
+  }
+
   isSelfLoopWithNoFeedback(outcome: Outcome): boolean | void {
-    if (outcome && typeof outcome === 'object' && this.stateName &&
+    let currentStateName = this.getActiveStateName();
+    if (outcome && typeof outcome === 'object' && currentStateName &&
       outcome.constructor.name === 'Outcome') {
-      return outcome.isConfusing(this.stateName);
+      return outcome.isConfusing(currentStateName);
     }
   }
 
   isSelfLoopThatIsMarkedCorrect(outcome: Outcome): boolean {
     if (!outcome) {
       return false;
     } else {
-      let currentStateName = this.stateName;
-
+      let currentStateName = this.getActiveStateName();
```

Comment:
use const whenever possible please

Reply:
Thanks @nikitaevg, I have made the suggested changes.

Reply:
When asked to change something try to think if this change is applicable to other pieces of code you touched.

You have two more lines which you touched and where you could change `let` to `const`.